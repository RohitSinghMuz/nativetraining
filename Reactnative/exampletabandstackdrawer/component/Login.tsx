import React, {Component} from 'react';
import {StyleSheet, Text, TextInput, TouchableOpacity, View} from 'react-native';

interface IProps {
  navigation?: any;
}
interface IState {
  username?:any,
  password?:any,
}

class Login extends Component<IProps, IState> {
  state={
    username:'',
    password:'',
  }
  render() {
    return (
      <View>

        <TextInput  placeholder='UserName' 
          style={styles.input}
           onChangeText={(newText)=>this.setState({username:newText})}
           value={this.state.username}
        />
        <TextInput  placeholder='Password'
             style={styles.input}
            onChangeText={(passText)=>this.setState({password:passText})}
            value={this.state.password}
        />


        <TouchableOpacity
          onPress={() => this.props.navigation.navigate('drawernav')}>
          <Text style={styles.btn}>Login </Text>
        </TouchableOpacity>
      </View>
    );
  }
}

export default Login;

const styles = StyleSheet.create({
  LoginHeading: {
    textAlign: 'center',
  },
    input: {
      height: 40,
      margin: 12,
      borderWidth: 1,
      padding: 10,
      marginHorizontal:20,
    },
    btn: {
      height: 40,
      width:200,
      marginHorizontal:100,
      marginVertical:20,
      borderWidth: 1,
      padding: 10,
      backgroundColor:'green',
      textAlign:'center',
    },
  
  
});
