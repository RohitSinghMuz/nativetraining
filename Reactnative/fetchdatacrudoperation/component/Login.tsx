import React, {Component} from 'react';
import {
  StatusBar,
  StyleSheet,
  Text,
  TextInput,
  TouchableOpacity,
  View,
} from 'react-native';
import AsyncStorage from '@react-native-async-storage/async-storage';
interface IProps {
  navigation?: any;
}
interface IState {
  email?: any;
  password?: any;
}

class Login extends Component<IProps, IState> {
  state = {
    email: '',
    password: '',
  };

  logindata = async () => {
    // console.log(this.state.email, this.state.password)
    try {
      const body = {
        "email": this.state.email,
        "password": this.state.password,
      };
      const options = {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(body),
      };
      const res = await fetch('https://reqres.in/api/login', options);
      const resJson = await res.json();
      // console.log(resJson);
      await AsyncStorage.setItem('token', JSON.stringify(resJson.token));

      if (res.status === 200) {
        this.props.navigation.navigate('bottomtab');
        console.log("Staus-------------",res.status)
        if (resJson?.token) {
          console.log(resJson.token);
        }
      } else {
        console.log('error');
      }
    } catch (error) {
      console.log(error);
    }
  };

  gettokendata = async () => {
    const getnewtoken: any = await AsyncStorage.getItem('token');
    const adata = (await getnewtoken) ? JSON.parse(getnewtoken) : [];
    console.log(adata, 'adata');
  };


  render() {
    return (
      <View  >
        {/* hidden={true}  */}
         <StatusBar  backgroundColor={'red'} />
        
        {/* <ImageBackground style={styles.bgimage} source={{uri:'https://images.pexels.com/photos/2486168/pexels-photo-2486168.jpeg?auto=compress&cs=tinysrgb&w=1600'}} /> */}
        <TextInput
          placeholder="Email"
          style={styles.input}
          onChangeText={newText => this.setState({email: newText})}
          value={this.state.email}
        />
        <TextInput
          placeholder="Password"
          style={styles.input}
          onChangeText={passText => this.setState({password: passText})}
          value={this.state.password}
        />

     

        <TouchableOpacity style={styles.btn}
          // style={{backgroundColor: 'grey'}}
          onPress={this.logindata}>
          <Text   style={styles.textdata}>Login </Text>
    
        </TouchableOpacity>
        {/* <TouchableOpacity
          // style={{backgroundColor: 'grey'}}
          onPress={this.gettokendata}>
          <Text style={styles.btn2}>Gettoken</Text>
        </TouchableOpacity> */}
      </View>
    );
  }
}

// "email": "eve.holt@reqres.in",
// "password": "cityslicka"

export default Login;

const styles = StyleSheet.create({
  LoginHeading: {
    textAlign: 'center',
  },
  input: {
    height: 40,
    margin: 12,
    borderWidth: 1,
    padding: 10,
    marginHorizontal: 20,
  },
  btn: {
    alignItems: 'center',
    justifyContent: 'center',
    marginHorizontal:100,
    textAlign:'center',
    marginVertical:30,
  },
  btn2: {
    alignItems: 'center',
    justifyContent: 'center',
    paddingVertical: 12,
    paddingHorizontal: 32,
    borderRadius: 4,
    elevation: 3,
    backgroundColor: 'green',
    marginHorizontal:100,
    textAlign:'center'
  },
  textdata: {
    textAlign:'center',
    padding:12,
    fontSize: 16,
    lineHeight: 21,
    fontWeight: 'bold',
    letterSpacing: 0.25,
   borderWidth:2,
   width:200,
   borderColor:'skyblue',
    color: 'black',
  },
  bgimage:{
    width:'100%',
    height:'100%',
    // zIndex:-3,
  }
});
