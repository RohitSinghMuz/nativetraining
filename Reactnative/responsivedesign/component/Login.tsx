import React, {Component} from 'react';
//@ts-ignore
// import { Icon } from 'react-native-vector-icons/MaterialIcons';
import AsyncStorage from '@react-native-async-storage/async-storage';

import {
  SafeAreaView,
  StyleSheet,
  Text,
  TextInput,
  TouchableOpacity,
  View,
} from 'react-native';
interface IProps {
  props?: any;
  navigation?:any
}
interface IState {
  Email: any;
  Password: any;
  data:any[],
}
class Login extends Component<IProps, IState> {
  state = {
    Email: '',
    Password: '',
    data:[],
  };

  addDataLogin=async()=>{
    this.props.navigation.navigate('tabnavigationdata')
    let arr={
      Email:this.state.Email,
      Password:this.state.Password,
    }
    this.setState({data:[...this.state.data,arr]})


    await AsyncStorage.setItem('logindata',JSON.stringify(this.state.data))

 
  }

  // getdata=async()=>{
  //   const getnewdata:any=await AsyncStorage.getItem('logindata')
  // const mydata= await  getnewdata?JSON.parse(getnewdata):[];
  // console.log(mydata,'mydata')
  //   }
  

  render() {
    return (
      <SafeAreaView style={styles.parentdiv}>
          {/* <TouchableOpacity  onPress={this.getdata}>
              <Text >GETDATA</Text>
            </TouchableOpacity> */}
        <View>
        {/* <Icon name="home" size={30} color="#900" /> */}
          <Text style={styles.heading}>Log in </Text>
          <TextInput
            placeholder="Email"
            style={styles.input}
            onChangeText={emailtext => this.setState({Email: emailtext})}
            value={this.state.Email}
          />
          <TextInput
            placeholder="Password"
            style={styles.input}
            secureTextEntry={true}
            onChangeText={passwordtext =>
              this.setState({Password: passwordtext})
            }
            value={this.state.Password}
          />

          <TouchableOpacity onPress={this.addDataLogin}>
            <Text style={styles.btnLogin}>Login to your account</Text>
          </TouchableOpacity>
          <View style={styles.signUpAccount}>
            <Text style={styles.accounttext}>I don't have account</Text>

            <TouchableOpacity onPress={()=>this.props.navigation.navigate("signup")}>
              <Text style={styles.signUpData}> Sign up</Text>
            </TouchableOpacity>
          </View>
        </View>
      </SafeAreaView>
    );
  }
}

export default Login;

const styles = StyleSheet.create({
  heading: {
    fontSize: 30,
    textAlign: 'left',
    marginHorizontal: 35,
  },
  input: {
    height: 48,
    margin: 12,
    padding: 10,
    marginHorizontal: 35,
    borderRadius: 5,
    backgroundColor: '#f2f2f2',
  },
  btnLogin: {
    height: 45,
    margin: 12,
    padding: 10,
    marginHorizontal: 35,
    backgroundColor: '#ff7269',
    paddingHorizontal: 20,
    color: 'white',
    fontSize: 16,
    borderRadius: 5,
  },
  parentdiv: {
    backgroundColor: 'white',

  },
  signUpAccount:{
  flexDirection:'row',
  fontSize:18,
  marginHorizontal: 35,

  },
  accounttext:{
    marginRight:6,
    fontSize:18,
    marginVertical:5,
  },
  signUpData:{
    fontSize:18,
    marginVertical:5,
    color:'#372EE5',
  }
});
