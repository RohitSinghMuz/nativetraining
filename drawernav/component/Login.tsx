import React, {useState} from 'react';
import {Formik, Form, Field} from 'formik';
import * as Yup from 'yup';
import {
  StyleSheet,
  Text,
  Touchable,
  TouchableOpacity,
  View,
} from 'react-native';
import {TextInput} from 'react-native-gesture-handler';

const SignupSchema = Yup.object().shape({
  firstname: Yup.string()
    .min(2, 'Too Short!')
    .max(50, 'Too Long!')
    .required('Required'),
  lastName: Yup.string()
    .min(2, 'Too Short!')
    .max(50, 'Too Long!')
    .required('Required'),
  email: Yup.string().email('Invalid email').required('Required'),
  password: Yup.string()
    .required('No password provided.')
    .min(6, 'Password is too short - should be 8 chars minimum.')
    .matches(/[a-zA-Z]/, 'Password can only contain Latin letters.'),
});

const Login = (props:any) => {
  const [email, setemail] = useState<any>('');
  const [password, setpassword] = useState<any>('');
  const [data, setdata] = useState<any>([]);
  const adddata = () => {
    let arr = {
      email,
      password,
    };
    setdata([...data, arr]);
    console.log(data, '***********************');
    setemail('');
    setpassword('');
    props.navigation.navigate('drawernav')
  };
  return (
    <View>
      <Formik
        initialValues={{
          email: '',
          password: '',
        }}
        validationSchema={SignupSchema}
        onSubmit={values => {
          // same shape as initial values
          console.log(values);
        }}>
        {({
          errors,
          touched,
          handleChange,
          handleBlur,
          handleSubmit,
          values,
        }) => (
          <View>
            <TextInput
              style={styles.input}
              placeholder="Enter Email"
              onBlur={handleBlur('email')}
              onChangeText={handleChange('email')}
            />
            {errors.email && touched.email ? <Text>{errors.email}</Text> : null}
            <TextInput
              style={styles.input}
              placeholder="Enter Password"
              onBlur={handleBlur('password')}
              onChangeText={handleChange('password')}
            />
            {errors.password && touched.password ? (
              <Text>{errors.password}</Text>
            ) : null}
            <TouchableOpacity onPress={adddata}>
              <Text style={styles.submitButtondata}>Submittdata</Text>
            </TouchableOpacity>

            <View style={styles.signupnav}>
              <Text style={{fontSize:25}}>If not Signup then </Text>
              <TouchableOpacity onPress={() =>props.navigation.navigate('signup')}>
                <Text style={{fontSize:25,color:'green'}}>Signup</Text>
              </TouchableOpacity>
            </View>
          </View>
        )}
      </Formik>
    </View>
  );
};

export default Login;

const styles = StyleSheet.create({
  signupnav: {
    display: 'flex',
    flexDirection:'row',
    textAlign:'center',
    margin: 15,
    height: 60,
    borderColor: '#f2f2f2',
    borderWidth: 1,
    fontSize: 25,
    justifyContent: 'center',
    paddingTop: 10,
  },
  container: {
    paddingTop: 23,
    backgroundColor: '#ffffff',
  },
  input: {
    margin: 15,
    marginTop: 20,
    height: 60,
    borderColor: 'grey',
    borderWidth: 1,
    display: 'flex',
    flexDirection: 'row',
    fontSize: 25,
  },

  submitButtondata: {
    margin: 15,
    height: 60,
    borderColor: '#f2f2f2',
    borderWidth: 1,
    display: 'flex',
    flexDirection: 'row',
    fontSize: 25,
    textAlign: 'center',
    justifyContent: 'center',
    backgroundColor: '#fe8279',
    paddingTop: 10,
  },

  submitButtonText: {
    color: 'white',
    textAlign: 'center',
  },
});
