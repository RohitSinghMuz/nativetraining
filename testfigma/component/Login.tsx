import React, {Component} from 'react';
import {Image, ImageBackground, StyleSheet, Text, View} from 'react-native';
import CheckBox from '@react-native-community/checkbox';
import {TextInput, TouchableOpacity} from 'react-native-gesture-handler';
import {SafeAreaView} from 'react-native-safe-area-context';
//@ts-ignore
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
//@ts-ignore
import Feather from 'react-native-vector-icons/Feather';
//@ts-ignore
import Entypo from 'react-native-vector-icons/Entypo';
//@ts-ignore
import image11 from './assetsimage/image11.png';
//@ts-ignore
import touchlogin from './assetsimage/touchlogin.png';
//@ts-ignore
import Subtract from './assetsimage/Subtract.png';

interface IProps {
  navigation?: any;
  props?: any;
}
interface IState {
  isChecked: boolean;
  inputemail: string;
  inputpassword: string;
  data: any[];
}
class Login extends Component<IProps, IState> {
  state = {
    isChecked: false,
    inputemail: '',
    inputpassword: '',
    data: [],
  };

  handleToggleCheckBox = () => {
    this.setState({isChecked: !this.state.isChecked});
  };

  addLoginData = () => {
    let arr = {
      inputemail: this.state.inputemail,
      inputpassword: this.state.inputpassword,
    };
    this.setState({data: [...this.state.data, arr]});
    console.log(this.state.data);
  };
  render() {
    return (
      <SafeAreaView>
        <View style={styles.container}>
          <View>
            <Text style={styles.loginHeading}>Login </Text>
          </View>
          <View>
            <Text style={styles.bysigning}>
              By signing in you are agreeing{' '}
            </Text>
          </View>
          <View style={styles.ourandterm}>
            <Text style={styles.ourtext}>our</Text>
            <Text style={styles.textterm}> Term and Privacy Policy</Text>
          </View>
          <View style={styles.loginsignup}>
            <TouchableOpacity
              testID="loginpage"
              // onPress={() => this.props.navigation.navigate('login')}
            >
              <Text style={styles.logintext}>Login</Text>
            </TouchableOpacity>
            <TouchableOpacity
              testID="registerpage"
              onPress={() => this.props.navigation.navigate('register')}>
              <Text style={styles.signuptext}>Register</Text>
            </TouchableOpacity>
          </View>
          <View style={styles.emailinput}>
            <MaterialCommunityIcons
              name="email-outline"
              size={30}
              color="grey"
              style={styles.emailicons}
            />
            <TextInput
              value={this.state.inputemail}
              testID="inputemail"
              style={styles.emailinputstyle}
              placeholder="Email Address"
              placeholderTextColor="gray"
              onChangeText={(emailinput: string) =>
                this.setState({inputemail: emailinput})
              }
            />
          </View>

          <View style={styles.emailinput}>
            <Feather
              name="lock"
              size={30}
              color="grey"
              style={styles.emailicons}
            />
            <TextInput
              value={this.state.inputpassword}
              testID="inputpassword"
              style={styles.emailinputstyle}
              placeholder="Password"
              placeholderTextColor="gray"
              onChangeText={(passwordtext: string) =>
                this.setState({inputpassword: passwordtext})
              }
            />
            <Entypo name="eye" size={30} color="grey" style={styles.eveicons} />
          </View>
          <View style={styles.forgotpasswordview}>
            <View style={styles.checkboxContainer}>
              {/* <CheckBox
                testID="checkboxID"
                style={styles.checkoboxitem}
                value={this.state.isChecked}
                onValueChange={this.handleToggleCheckBox}
              /> */}
              <Text style={styles.label}>Remember Password</Text>
            </View>
            <View>
              <Text style={styles.forgotpasswordtext}>Forgot Password</Text>
            </View>
          </View>
          <View>
            <TouchableOpacity
              testID="loginID"
              style={styles.logindesign}
              onPress={this.addLoginData}>
              <Text style={styles.logintextdata}>Login</Text>
            </TouchableOpacity>
          </View>
          <View>
            <Text style={styles.connectwith}>or connect with</Text>
          </View>
          <View style={styles.socialicons}>
            <Entypo
              name="facebook-with-circle"
              size={30}
              color="grey"
              style={styles.facebookicons}
            />
            <Entypo
              name="instagram-with-circle"
              size={30}
              color="grey"
              style={styles.facebookicons}
            />
            <Entypo
              name="pinterest-with-circle"
              size={30}
              color="grey"
              style={styles.pinteresticons}
            />
            <Entypo
              name="linkedin-with-circle"
              size={30}
              color="grey"
              style={styles.linkedinicons}
            />
          </View>

          <View>
            <Image style={styles.image11} source={image11} />
          </View>

          <View style={styles.halfeclipse}>
            <Image style={styles.subimage} source={Subtract} />
            <Image style={styles.touchlogin} source={touchlogin} />
            <Text style={styles.touchtext}>Login with touch</Text>
          </View>
        </View>
      </SafeAreaView>
    );
  }
}

export default Login;

const styles = StyleSheet.create({
  loginHeading: {
    fontFamily: 'majalla',
    marginTop: 40,
    fontSize: 40,
    lineHeight: 56,
    textAlign: 'center',
    color: '#000000',
    fontWeight: '400',
  },
  container: {
    backgroundColor: '#FFFFFF',
    height: '100%',
  },
  bysigning: {
    fontFamily: 'majalla',
    fontSize: 25,
    fontWeight: '400',
    lineHeight: 35,
    letterSpacing: 0,
    textAlign: 'center',
    color: '#6B5E5E',
    fontStyle: 'normal',
  },
  ourandterm: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
  },
  textterm: {
    fontFamily: 'majalla',
    fontSize: 25,
    fontWeight: '400',
    lineHeight: 35,
    letterSpacing: 0,
    textAlign: 'center',
    color: '#0386D0',
  },
  ourtext: {
    fontFamily: 'majalla',
    fontSize: 25,
    fontWeight: '400',
    lineHeight: 35,
    letterSpacing: 0,
    textAlign: 'center',
    color: '#6B5E5E',
    fontStyle: 'normal',
  },
  loginsignup: {
    display: 'flex',
    justifyContent: 'center',
    flexDirection: 'row',
    textAlign: 'center',
  },
  logintext: {
    fontFamily: 'majalla',
    fontSize: 22,
    fontWeight: '400',
    lineHeight: 35,
    letterSpacing: 0,
    textAlign: 'center',
    margin: 10,
    borderBottomWidth: 1.5,
    color: '#0386D0',
  },
  signuptext: {
    fontFamily: 'majalla',
    fontSize: 22,
    fontWeight: '400',
    lineHeight: 35,
    letterSpacing: 0,
    textAlign: 'center',
    margin: 10,
  },
  emailinput: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'flex-start',
    marginLeft: 35,
    borderBottomColor: '#A6A6A6',
    borderBottomWidth: 1.5,
    marginRight: 40,
    marginBottom: 20,
  },
  emailinputstyle: {
    fontFamily: 'majalla',
    fontSize: 22,
    marginLeft: 10,
  },
  emailicons: {
    marginTop: 10,
  },
  eveicons: {
    marginLeft: 120,
    marginTop: 8,
  },
  logintextdata: {
    fontSize: 25,
    textAlign: 'center',
    color: '#FFFFFF',
    fontFamily: 'majalla',
    lineHeight: 35,
    letterSpacing: 0,
  },
  checkboxContainer: {
    flexDirection: 'row',
    marginBottom: 20,
  },
  checkoboxitem: {
    marginTop: 5,
  },
  checkbox: {
    alignSelf: 'center',
    margin: 18,
  },
  label: {
    margin: 8,
    fontFamily: 'majalla',
    fontSize: 14,
    fontWeight: '400',
    letterSpacing: 0,
    textAlign: 'left',
  },
  forgotpasswordview: {
    display: 'flex',
    flexDirection: 'row',
    marginHorizontal: 28,
  },
  forgotpasswordtext: {
    fontFamily: 'majalla',
    marginLeft: 20,
    marginTop: 10,
    fontSize: 14,
    color: '#0386D0',
  },
  logindesign: {
    marginHorizontal: 35,
    backgroundColor: '#0386D0',
    padding: 12,
    borderRadius: 5,
    textAlign: 'center',
    color: '#FFFFFF',
    fontSize: 18,
  },
  connectwith: {
    marginHorizontal: 35,
    fontSize: 25,
    textAlign: 'center',
    marginTop: 12,
    margin: 7,
    fontFamily: 'majalla',
  },
  facebookicons: {
    color: '#3B5998',
    fontSize: 45,
    margin: 7,
  },
  pinteresticons: {
    color: '#CB2027',
    fontSize: 45,
    margin: 7,
  },
  linkedinicons: {
    color: '#007AB9',
    fontSize: 45,
    margin: 7,
  },
  socialicons: {
    display: 'flex',
    justifyContent: 'center',
    flexDirection: 'row',
  },
  image11: {
    height: 100,
    width: '100%',
    objectFit: 'fill',
    zIndex: 2,
  },
  touchlogin: {
    marginTop: -38,
    marginBottom: 15,
    left: '40%',
  },

  halfeclipse: {
    width: '100%',
    height: 330,
    backgroundColor: '#469FD1',
    borderTopRadius: '50%',
    zIndex: 999999999,
  },
  subimage: {
    marginTop: -100,
    width: '100%',
    height: 100,
  },
  touchtext: {
    textAlign: 'center',
    fontSize: 25,
    marginBottom: 5,
    fontFamily: 'majalla',
    color: '#FFFFFF',
  },
});
