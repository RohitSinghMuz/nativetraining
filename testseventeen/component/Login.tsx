import {Component, ReactNode} from 'react';
import {Formik, Form, Field} from 'formik';
import * as Yup from 'yup';
import {
  StyleSheet,
  TextInput,
  View,
  Text,
  TouchableOpacity,
} from 'react-native';
import {Chip} from 'react-native-paper';
//@ts-ignore
import AntDesign from 'react-native-vector-icons/AntDesign';

//@ts-ignore
import Entypo from 'react-native-vector-icons/Entypo';

interface IProps {
  navigation?: any;
  props?: any;
}
interface IState {
  username: any;
  password: any;
  data: any[];
}

const SignupSchema = Yup.object().shape({
  username: Yup.string()
    .min(4, 'Too Short!')
    .max(50, 'Too Long!')
    .required('Required'),
  lastName: Yup.string()
    .min(2, 'Too Short!')
    .max(50, 'Too Long!')
    .required('Required'),
  email: Yup.string().email('Invalid email').required('Required'),
  password: Yup.string()
    .required('No password provided.')
    .min(6, 'Password is too short - should be 8 chars minimum.')
    .matches(/[a-zA-Z]/, 'Password can only contain Latin letters.'),
});

class Login extends Component<IProps, IState> {
  state = {
    password: '',
    username: '',
    data: [],
  };

  addsignindata = () => {
    let objdata = {
      usename: this.state.username,
      password: this.state.password,
    };
    this.setState({data: [...this.state.data, objdata]});
    this.props.navigation.navigate('signup');
  };

  submittdata = () => {
    this.props.navigation.navigate('signup');
  };
  render() {
    return (
      <View>
        <View style={styles.headerdesign}>
          <Text style={styles.nutriDesign}>Nutri</Text>
          <Text style={styles.izydesign}>izy</Text>
        </View>
        <View style={styles.logincontent}>
          <View style={styles.letssignParent}>
            <View>
              <Text style={styles.signyou}>Let's Sign you in</Text>
              <Text style={styles.youMissed}>you have been Missed!</Text>
            </View>
          </View>

          {/* <View>
            <Formik
              initialValues={{
                username: '',
                password: '',
              }}
              validationSchema={SignupSchema}
              onSubmit={(values, actions) => {
                //  console.log(values);
                // actions.setSubmitting(false);
              }}>
              {({
                errors,
                touched,
                handleChange,
                handleBlur,
                handleSubmit,
                values,
              }) => (
                <View>
                  <View>
                    <Text style={styles.emaillabel}>Your Email / Username</Text>
                    <View style={styles.inputContainer}>
                      <TextInput
                        style={styles.inputField}
                        value={values.username}
                        placeholder="John"
                        onBlur={handleBlur('username')}
                        onChangeText={handleChange('username')}
                      />

                      {errors.username && touched.username ? (
                        <Text style={{fontSize: 20}}>{errors.username}</Text>
                      ) : null}
                    </View>
                  </View>

                  <View>
                    <Text style={styles.passwordlabel}>Your Password</Text>
                    <View style={styles.inputContainer}>
                      <TextInput
                        style={styles.inputField}
                        value={values.password}
                        placeholder="Password"
                        onBlur={handleBlur('password')}
                        onChangeText={handleChange('password')}
                      />

                      {errors.password && touched.password ? (
                        <Text style={{fontSize: 20}}>{errors.password}</Text>
                      ) : null}
                    </View>

                    <View
                      style={{
                        display: 'flex',
                        justifyContent: 'flex-end',
                        flexDirection: 'row',
                      }}>
                      <Text style={styles.forgotPassword}>
                        Forgot Password?
                      </Text>
                    </View>
                 
                  </View>
                </View>
              )}
            </Formik>
          </View> */}

          <View>
            <Text style={styles.emaillabel}>Your Email / Username</Text>

            <View style={styles.inputContainer}>
              <TextInput
                style={styles.inputField}
                placeholder="Stuart Reichel"
                autoCapitalize="none"
                autoCorrect={false}
                secureTextEntry
                value={this.state.username}
                enablesReturnKeyAutomatically
                onChangeText={usernametext =>
                  this.setState({username: usernametext})
                }
              />
            </View>
          </View>
          <View>
            <Text style={styles.passwordlabel}>Your Password</Text>

            <View style={styles.inputContainer}>
              <TextInput
                style={styles.inputField}
                placeholder="Password"
                autoCapitalize="none"
                autoCorrect={false}
                secureTextEntry
                value={this.state.password}
                enablesReturnKeyAutomatically
                onChangeText={passwordtext =>
                  this.setState({password: passwordtext})
                }
              />
            </View>
          </View>
          <View>
            <TouchableOpacity onPress={this.addsignindata}>
              <Text style={styles.signindesign}>Sign Up</Text>
            </TouchableOpacity>
          </View>

          <View style={styles.socialIconsParent}>
            <TouchableOpacity
              onPress={() => this.props.navigation.navigate('googleauth')}
              style={styles.iconsdesign}>
              <Text style={{textAlign: 'center'}}>
                <AntDesign name="google" color={'red'} size={25} />
              </Text>
            </TouchableOpacity>
            <Text style={styles.iconsdesign}>
              <Entypo name="facebook-with-circle" color={'#5281eb'} size={25} />
            </Text>
          </View>

          <View style={styles.signupdesignView}>
            <Text style={styles.dontAccounttext}>Don't have a Account?</Text>
            <TouchableOpacity
              onPress={() => this.props.navigation.navigate('signup')}>
              <Text style={styles.signupdesigntext}>Sign Up</Text>
            </TouchableOpacity>
          </View>

          <View style={styles.termandcondition}>
            <Text style={styles.continue}>
              By continuing you agree Nuyrizy's
            </Text>
            <TouchableOpacity>
              <Text style={styles.termservice}>Terms of</Text>
            </TouchableOpacity>
          </View>
          <View>
            <Text style={styles.services}>services & Privacy Policy</Text>
          </View>
        </View>
      </View>
    );
  }
}

export default Login;

const styles = StyleSheet.create({
  headerdesign: {
    backgroundColor: '#f6f7fc',
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'center',
    marginTop: 10,
    padding: 20,
  },
  nutriDesign: {
    color: '#000000',
    fontSize: 25,
    fontWeight: 'bold',
  },
  izydesign: {
    color: '#47ae62',
    fontSize: 25,
    fontWeight: 'bold',
  },
  letssignParent: {
    backgroundColor: '#ffffff',
  },

  signyou: {
    marginHorizontal: 20,
    fontSize: 20,
    color: '#000000',
    paddingTop: 10,
    fontWeight: '400',
  },
  youMissed: {
    fontSize: 30,
    marginHorizontal: 20,
    color: '#2d292a',
    fontWeight: 'bold',
  },
  inputContainer: {
    backgroundColor: 'white',
    width: '90%',
    marginHorizontal: 20,
    borderBottomWidth: 2,
    borderColor: '#79c38a',
  },
  inputField: {
    padding: 14,
    fontSize: 22,
    width: '90%',
  },
  emaillabel: {
    fontSize: 16,
    marginHorizontal: 20,
    marginTop: 20,
    fontWeight: '400',
  },
  passwordlabel: {
    fontSize: 16,
    marginHorizontal: 20,
    marginTop: 20,
    fontWeight: '400',
  },
  forgotPassword: {
    marginRight: 20,
    marginTop: 20,
    fontSize: 20,
    fontWeight: '500',
  },
  signindesign: {
    textAlign: 'center',
    backgroundColor: '#33a854',
    marginHorizontal: 20,
    color: '#ffffff',
    fontWeight: 'bold',
    padding: 15,
    borderRadius: 35,
    fontSize: 20,
    marginTop: 20,
  },
  socialIconsParent: {
    display: 'flex',
    flexDirection: 'row',
    marginHorizontal: 20,
    marginVertical: 20,
    justifyContent: 'space-between',
  },
  iconsdesign: {
    width: '48%',
    borderWidth: 1,
    padding: 10,
    borderColor: 'grey',
    borderRadius: 28,
    textAlign: 'center',
  },
  signupdesignView: {
    display: 'flex',
    flexDirection: 'row',
    marginHorizontal: 40,
    paddingTop: 10,
    paddingBottom: 40,
  },
  dontAccounttext: {
    fontSize: 20,
    fontWeight: '500',
    color: '#616267',
  },
  signupdesigntext: {
    fontSize: 20,
    marginLeft: 4,
    color: 'black',
    fontWeight: '700',
  },

  termandcondition: {
    marginHorizontal: 20,
    paddingTop: 20,
    borderTopWidth: 1,
    borderColor: '#616267',
    display: 'flex',
    justifyContent: 'flex-start',
    flexDirection: 'row',
  },
  continue: {
    fontSize: 18,
  },
  termservice: {
    fontSize: 16,
    fontWeight: 'bold',
  },
  services: {
    textAlign: 'center',
    fontWeight: 'bold',
    padding: 6,
    fontSize: 18,
  },
  logincontent: {
    backgroundColor: '#ffffff',
    borderRadius: 30,
  },
});
