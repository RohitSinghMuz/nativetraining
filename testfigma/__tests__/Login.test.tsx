import {NavigationContainer} from '@react-navigation/native';
import {createStackNavigator} from '@react-navigation/stack';
import {render, fireEvent, act} from '@testing-library/react-native';
import renderer from 'react-test-renderer';
import {Button} from 'react-native';
import Login from '../component/Login';
import {clockRunning} from 'react-native-reanimated';

jest.mock('react-native-vector-icons/Feather', () => () => <></>);
jest.mock('react-native-vector-icons/MaterialCommunityIcons', () => () => (
  <></>
));

jest.mock('react-native-vector-icons/Entypo', () => () => <></>);

jest.mock('@react-navigation/native-stack', () => ({
  createNativeStackNavigator: jest.fn(),
}));
describe('signup testing', () => {
  // it('renders correctly', () => {
  //     const tree = render(<Signup />);
  //     expect(tree).toMatchSnapshot();
  //   });
  it(`renders correctly`, () => {
    const tree = renderer.create(<button>Login</button>);
    expect(tree).toMatchSnapshot();
  });

  it('login test Cases', async () => {
    // Add any additional assertions here to verify the component's behavior

    const {getByTestId} = render(<Login navigation={{navigate: jest.fn()}} />);
    const inputemail: any = getByTestId('inputemail');

    const inputpassword: any = getByTestId('inputpassword');

    await act(async () => {
      await fireEvent.changeText(inputemail, 'abc@gmail.com');

      await fireEvent.changeText(inputpassword, 'abc345#');
    });

    console.log('====================================');
    console.log('----- value', inputemail.props.value);

    expect(inputemail.props.value).toBe('abc@gmail.com');
    expect(inputpassword.props.value).toBe('abc345#');
  });

  it('validates successfully', async () => {
    const {getByTestId} = render(<Login />);
    //  navigation={{navigate: jest.fn()}}
    const inputemail: any = getByTestId('inputemail');

    const inputpassword: any = getByTestId('inputpassword');
    await act(async () => {
      await fireEvent.changeText(inputemail, 'abc@gmail.com');
      await fireEvent.changeText(inputpassword, 'abc345#');
    });

    const submitBtn = getByTestId('loginID');
    await act(async () => {
      await fireEvent.press(submitBtn);
    });
  });

  it('navigatetologinpage', async () => {
    const handleSubmit = jest.fn();
    const {getByTestId} = render(
      <Login handleSubmit={handleSubmit} navigation={{navigate: jest.fn()}} />,
    );
    const LoginButton = getByTestId('loginpage');
    fireEvent.press(LoginButton);
    expect(handleSubmit).toBeCalledTimes;
  });

  it('navigatetoregisterpage', async () => {
    const handleSubmit = jest.fn();
    const {getByTestId} = render(
      <Login handleSubmit={handleSubmit} navigation={{navigate: jest.fn()}} />,
    );
    const RegisterPage = getByTestId('registerpage');
    fireEvent.press(RegisterPage);
    expect(handleSubmit).toBeCalledTimes;
  });

  it('Testing Checkbox Input', async () => {
    const {getByTestId} = render(<Login />);
    const CheckBoxclick = getByTestId('checkboxID');
    fireEvent.press(CheckBoxclick);
    // RegisterPage.props.children.props.onValueChange('test');
    expect(CheckBoxclick).toBeCalledTimes;
  });

  // it('Testing Checkbox Input', async () => {
  //   const {getByTestId} = render(<Login />);
  //   const CheckBoxclick = getByTestId('checkboxID');
  //   // Execute the click event of the checkbox
  //   fireEvent.press(CheckBoxclick);
  //   expect(CheckBoxclick).toBeCalledWith();
  //   expect(getByTestId('checkboxID')).toBeDefined();

  //   // Execute the click event again
  //   fireEvent.press(CheckBoxclick);
  //   expect(CheckBoxclick).not.toBeCalledWith();
  //   expect(getByTestId('checkboxID')).not.toBeDefined();
  // });
});
