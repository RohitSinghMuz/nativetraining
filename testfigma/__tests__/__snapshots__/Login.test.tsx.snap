// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`signup testing renders correctly 1`] = `
<button>
  Login
</button>
`;
